lib/m2ee/munin.py

```diff
@@ -229,21 +231,185 @@ def get_stats_from_runtime(client, config):
+            "CodeHeap 'non-profiled nmethods'",
+        ),
+        "permanent": ("Metaspace",),
+        "eden": ("Eden Space",),
+        "survivor": ("Survivor Space",),
+        "tenured": ("Tenured Gen",),
+    }
+
+    if java_version == 8:
+        pool_mapping = java_8_mapping
+    elif java_version == 11:
+        pool_mapping = java_11_mapping
+    else:
+        # Why raise, instead of trying and "guess" based on known JVM/JREs?
+        # Because the mapping has changed in every supported JRE. Better to
+        # fail fast, so that mapping can be confirmed during testing of a new
+        # JVM/JRE version.
+        raise NotImplementedError(
+            "Java version {} does not yet have a memorypool mapping in "
+            "m2ee tools".format(java_version)
+        )
+
+    # Transform memorypools from a list of dicts, to a dict of memory usages,
+    # which is more what we want. Additionally ensure we use a standard pool
+    # name. ie. For "PS Eden Space" use "Eden Space"
+    memory_pools_dict = {
+        _standard_pool_name(f["name"]): f["usage"]
+        for f in runtime_memory_pools
+    }
+
+    output_stats = {}
+    for our_memory_type, pool_names in pool_mapping.items():
+        try:
+            total = sum(
+                [memory_pools_dict[pool_name] for pool_name in pool_names]
+            )
+        except KeyError as exc:
+            got_fields = list(memory_pools_dict.keys())
+            required_fields = list(itertools.chain(*pool_mapping.values()))
+            logger.error(
+                "Collecting JVM memory pool stats failed. Memory pool "
+                "output did not match expected output. Needed: %s. Got: %s",
+                required_fields,
+                got_fields,
+            )
+            raise RuntimeError(
+                "Unable to collect JVM memory pool stats. "
+                "Please contact support!"
+            ) from exc
+        output_stats[our_memory_type] = total
+
+    return output_stats
+
+
+def _standard_pool_name(pool_name):
+    """Return a standard memory pool name.
+
+    The memory pool names could vary based on the garbage collector enabled.
+    This function returns a standard name we could refer to.
+
+    Available collectors :
+    https://docs.oracle.com/en/java/javase/11/gctuning/available-collectors.html
+
+    Here is an old gist listing the variations in the names:
+    https://gist.github.com/szegedi/1474365
+
+    """
+
+    # mapping of standard memory pool name to all known names.
+    pool_name_aliases = {
+        "Eden Space": ("PS Eden Space", "Par Eden Space", "G1 Eden Space",),
+        "Survivor Space": (
+            "PS Survivor Space",
+            "Par Survivor Space",
+            "G1 Survivor Space",
+        ),
+        "Tenured Gen": ("PS Old Gen", "CMS Old Gen", "G1 Old Gen",),
+    }
+
+    for standard_name, valid_names in pool_name_aliases.items():
+        for name in valid_names:
```

Comment:
I think that this would be slightly easier to read (at least for me) if we reversed the test, and also renamed `pool_name` to `given_pool_name`, i.e.:

```
for name in valid_names:
    if name == given_pool_name:
        return standard_name
```

What do you think?

Reply:
Sure. I'll update it